### PR TITLE
Adding Swift WebAssembly support

### PR DESCRIPTION
**Pull request by fjtrujy**
Hello,
This PR basically just adds Swift WebAssemby support to the `SwiftPrettyPrint`.

I have added the compilation step into the CI, however, testing for web assembly is not as easy as it should be yet, so this is why I couldn't add the test stage in the CI.

Additionally, I have removed the usage of legacy Xcode 13.1

Thanks

### CODE REVIEW

**Comment by fjtrujy** [comment]
Hello @YusukeHosonuma ,
I think CodeCov is not interpreting properly the precompilation conditions.

The ignored properties were needed to make it compile for `Swift Web Assembly toolchain` because they aren't available there.

Thanks

File: Sources/Core/Log/LogOutputStream.swift [hunk 1]
```diff
@@ -22,9 +23,11 @@
                     print("[SwiftPrettyPrint] Failed to create directory. (\(directoryPath))")
                 }
             }
+            #endif
         }
 
         func write(_ string: String) {
+#if !os(WASI)
```
Comment (sahara-ooga) [inline Sources/Core/Log/LogOutputStream.swift]: [imo]

It is preferable to align the indentation with the other lines:

```suggestion
            #if !os(WASI)
```

File: Sources/Core/Log/LogOutputStream.swift [hunk 2]
```diff
@@ -37,6 +40,7 @@
                 }
                 _ = output.write(bytes, maxLength: data.count)
             }
+#endif
```
Comment (sahara-ooga) [inline Sources/Core/Log/LogOutputStream.swift]: [imo]

It is preferable to align the indentation with the other lines:

```suggestion
            #endif
```

**Review by sahara-ooga — APPROVED** [review]
Thank you for interesting addition.
There are some minor suggestions but I approve this PR!

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -46,18 +46,23 @@ jobs:
...
     runs-on: macos-latest
 
     steps:
     - uses: actions/checkout@v2
 
-    - name: Select Xcode 11.3.1
-      run: |
-        sudo xcode-select -s '/Applications/Xcode_11.3.1.app/Contents/Developer'
-        xcodebuild -version
-
```
Comment (sahara-ooga) [inline .github/workflows/main.yml:60]: 👍🏻

**Comment by fjtrujy** [comment]
Hello @YusukeHosonuma, your suggestion have been applied

**Comment by fjtrujy** [comment]
Hi again @YusukeHosonuma!
I think that in your recent commits you have pushed something that is not passing the CI.

**Pull request merged**